.github/workflows/integrational-tests.yml

```diff
@@ -0,0 +1,113 @@
+# This workflow will build a docker container, publish it to Google Container Registry, and deploy it to GKE when a release is created
+#
+# To configure this workflow:
+#
+# 1. Ensure that your repository contains the necessary configuration for your Google Kubernetes Engine cluster, including deployment.yml, kustomization.yml, service.yml, etc.
+#
+# 2. Set up secrets in your workspace: GKE_PROJECT with the name of the project and GKE_SA_KEY with the Base64 encoded JSON service account key (https://github.com/GoogleCloudPlatform/github-actions/tree/docs/service-account-key/setup-gcloud#inputs).
+#
+# 3. Change the values for the GKE_ZONE, GKE_CLUSTER, IMAGE, and DEPLOYMENT_NAME environment variables (below).
+#
+# For more support on how to run the workflow, please visit https://github.com/GoogleCloudPlatform/github-actions/tree/master/example-workflows/gke
+
+name: Test on GKE
+on: push
+
+env:
+  PROJECT_ID: ${{ secrets.GKE_PROJECT }}
+  GKE_CLUSTER: guides-tests
+  GKE_ZONE: europe-west1-b
+  ACTIONS_ALLOW_UNSECURE_COMMANDS: true
+
+jobs:
+  run-tests:
+    name: Run Intergrational tests on GKE
+    runs-on: ubuntu-latest
+    continue-on-error: true
+
+    outputs:
+      master_ip: ${{ steps.master-ip.outputs.ip }}
+      api_token: ${{ steps.secrets.outputs.apiToken }}
+      ca_certificate: ${{ steps.secrets.outputs.caCertificate }}
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+
+
+    - name: Set up JDK 1.8
+      uses: actions/setup-java@v1
+      with:
+        java-version: 1.8
+
+    # Setup gcloud CLI
+    - uses: GoogleCloudPlatform/github-actions/setup-gcloud@0.1.3
+      with:
+        service_account_key: ${{ secrets.GKE_SA_KEY }}
+        project_id: ${{ env.PROJECT_ID }}
+        export_default_credentials: true 
+
+    - run: |-
+        gcloud components install kubectl
+
+    - run: |-
+        gcloud container clusters create "$GKE_CLUSTER" --zone="$GKE_ZONE" --project="$PROJECT_ID" --machine-type=n1-standard-2 --num-nodes=2
+
+    - run: |-
+        gcloud container clusters get-credentials --region="$GKE_ZONE" "$GKE_CLUSTER"
+
+    - run: |-
+        kubectl config use-context $(kubectl config get-contexts -o name)
+      continue-on-error: true
+
+    - name: Apply RBAC
+      run: |-
+        kubectl apply -f rbac.yaml
+      continue-on-error: true
```

Comment:
Why do we want them to continue? For instance, Isn't the next run going to fail if `kubectl apply -f rbac.yaml` fails?

Reply:
Initially, it was done for clean-up reasons, so that when this step fails it will still proceed to the clean-up job. But since not we have a separate step that runs on failure it not needed anymore
Will remove all of `continue-on-error`s

Reply:
Got it.
Let me ask that also: What if clean up on failure job fails? See this job for instance: https://github.com/hazelcast-guides/kubernetes-external-client/runs/1422958675

Cleanup of failure fails since:
```
ResponseError: code=400, message=Operation operation-1605773859633-b29c1e91 
is currently creating cluster guides-tests. Please wait and try again once it is done.
```
Does this cluster still exist after the action fails?

Reply:
For this particular case, the cleanup failed because there were two actions running at the same time, so the cleanup tried to delete the cluster that was being created.
One way to overcome it is to make the cluster name unique by appending a timestamp to the name, for example.
But if it fails for any other reason the cluster might remain running. And I didn't come up with any ideas on how to overcome this scenario